features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/metric/interactiontonextview/InteractionToNextViewMetricResolver.kt

```diff
@@ -0,0 +1,116 @@
+        lastViewCreatedTimestamps[viewId] = timestamp
+        purgeOldEntries()
+    }
+
+    fun onActionSent(context: InternalInteractionContext) {
+        if (ingestionValidator.validate(context)) {
+            lastInteractions[context.viewId] = context
+        }
+        purgeOldEntries()
+    }
+
+    @Suppress("ReturnCount")
+    fun resolveMetric(viewId: String): Long? {
+        purgeOldEntries()
+        val currentViewCreatedTimestamp = lastViewCreatedTimestamps[viewId]
+        if (currentViewCreatedTimestamp == null) {
+            internalLogger.log(
+                InternalLogger.Level.WARN,
+                InternalLogger.Target.MAINTAINER,
+                { "[ViewNetworkSettledMetric] The view was not yet created for this viewId:$viewId" }
+            )
+            return null
+        }
+        val lastPrevViewInteraction = resolveLastInteraction(viewId, currentViewCreatedTimestamp)
+        if (lastPrevViewInteraction != null) {
+            val difference = currentViewCreatedTimestamp - lastPrevViewInteraction.eventCreatedAtNanos
+            if (difference > 0) {
+                return difference
+            } else {
+                internalLogger.log(
+                    InternalLogger.Level.WARN,
+                    InternalLogger.Target.MAINTAINER,
+                    {
+                        "[ViewNetworkSettledMetric] The difference between the last interaction " +
+                            "and the current view is negative for viewId:$viewId"
+                    }
+                )
+                return null
+            }
+        }
+        internalLogger.log(
+            InternalLogger.Level.WARN,
+            InternalLogger.Target.MAINTAINER,
+            { "[ViewNetworkSettledMetric] No previous interaction found for this viewId:$viewId" }
+        )
+        return null
+    }
+
+    private fun resolveLastInteraction(viewId: String, currentViewCreatedTimestamp: Long): InternalInteractionContext? {
+        val currentViewIdIndex = lastViewCreatedTimestamps.keys.indexOf(viewId)
+        val previousViewId = lastViewCreatedTimestamps.keys.elementAtOrNull(currentViewIdIndex - 1)
+        if (previousViewId != null) {
+            lastInteractions[previousViewId]?.let {
+                val context = PreviousViewLastInteractionContext(
+                    it.actionType,
+                    it.eventCreatedAtNanos,
+                    currentViewCreatedTimestamp
+                )
+                if (lastInteractionIdentifier.validate(context)) {
+                    return it
+                }
+            }
+        }
+        return null
+    }
+
+    private fun purgeOldEntries() {
+        while (lastInteractions.entries.size > MAX_ENTRIES) {
+            @Suppress("UnsafeThirdPartyFunctionCall")
+            // we make sure the collection is never empty
+            lastInteractions.entries.remove(lastInteractions.entries.first())
```

Comment:
what's the reason behind removing the first entry of a hashmap?

Reply:
we are purging the data basically making sure we are only keeping the last (newly added) values.

Reply:
OK, I see it's a `LinkedHashMap` so it makes sense